java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -336,9 +356,22 @@ protected final <Z> Z createProxy(final Decorated<Z> decorated, Class<Z> clazz)
       .getLoaded()
       .asSubclass(clazz);
 
+    if (clazz.isInterface()
+        || this.targetWebDriverConstructorArgs.length == 0
+        || this.targetWebDriverConstructorArgTypes.length == 0
+        ) {
```

Comment:
```suggestion
    if (clazz.isInterface()) {
```
the code below should work perfectly for constructors without args, so no need to add these checks